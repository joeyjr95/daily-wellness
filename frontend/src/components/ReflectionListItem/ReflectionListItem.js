import React, { Component } from 'react'
import moment from 'moment'
import WellnessContext from '../../contexts/WellnessContext'


export default class ReflectionListItem extends Component{
    static contextType = WellnessContext
    static defaultProps ={
        handleClickEdit: () => {}
    }

    handleDelete = e =>{
        this.context.handleDeleteReflection(this.props.id)
           
    }
    handleEdit = e =>{
        this.context.setReflection(this.props)
        this.props.handleClickEdit()
        
    }
  
    render(){
        
        return(
            <div >
                <header className='ReflectionListItem_header'>
                  <h2> {moment(this.props.date_created).format('MM/DD/YYYY')} </h2>
                </header>
                <span className='ReflectionListItem_mentalrating'>
           <div> mental rating:  {this.props.mental_rating}  </div>
        </span>
        <span className='ReflectionListItem_mentalcontent'>
           <div> mental content:</div>
           <p>{this.props.mental_content}</p>
        </span>
        <span className='ReflectionListItem_physicalrating'>
            <div> physical rating:  {this.props.physical_rating} </div>
        </span>
        <span className='ReflectionListItem_physicalcontent'>
           <div> physical content:  </div>
           <p>{this.props.physical_content}</p>
        </span>
            <div className="buttons">
            <button onClick={this.handleDelete}>
                Delete
            </button>
            <button onClick={this.handleEdit}>  
                Edit
            </button>
            </div>


            </div>

        )

    }

}

