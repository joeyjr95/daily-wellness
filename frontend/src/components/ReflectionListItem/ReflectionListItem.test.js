import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ReflectionListItem from './ReflectionListItem'

describe(`ReflectionListItem component`, () => {
  const props = {
    id: 12,
    date_created: "2018-08-15T23:00:00.000Z",
    mental_rating: 5,
    mental_content: "stuff",
    physical_rating: 2,
    physical_content: "stuff",
    handleClickEdit: () => {
        this.props.history.push(`/edit`)
        }
  }

  it('renders a ReflectionListItem by default', () => {
    const wrapper = shallow(<ReflectionListItem />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the ReflectionListItem given props', () => {
    const wrapper = shallow(<ReflectionListItem {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})