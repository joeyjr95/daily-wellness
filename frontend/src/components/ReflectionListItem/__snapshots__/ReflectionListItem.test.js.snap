// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReflectionListItem component renders a ReflectionListItem by default 1`] = `
<div
  className="ReflectionListItem"
>
  <header
    className="ReflectionListItem_header"
  >
    <h2>
       
      10/23/2019
       
    </h2>
  </header>
  <span
    className="ReflectionListItem_mentalrating"
  >
    <div>
       mental rating:  
        
    </div>
  </span>
  <span
    className="ReflectionListItem_mentalcontent"
  >
    <div>
       mental content:  
        
    </div>
  </span>
  <span
    className="ReflectionListItem_physicalrating"
  >
    <div>
       physical rating:  
       
    </div>
  </span>
  <span
    className="ReflectionListItem_physicalcontent"
  >
    <div>
       physical content:  
        
    </div>
  </span>
  <button
    onClick={[Function]}
  >
    Delete
  </button>
  <button
    onClick={[Function]}
  >
    Edit
  </button>
</div>
`;

exports[`ReflectionListItem component renders the ReflectionListItem given props 1`] = `
<div
  className="ReflectionListItem"
>
  <header
    className="ReflectionListItem_header"
  >
    <h2>
       
      08/15/2018
       
    </h2>
  </header>
  <span
    className="ReflectionListItem_mentalrating"
  >
    <div>
       mental rating:  
      5
        
    </div>
  </span>
  <span
    className="ReflectionListItem_mentalcontent"
  >
    <div>
       mental content:  
      stuff
        
    </div>
  </span>
  <span
    className="ReflectionListItem_physicalrating"
  >
    <div>
       physical rating:  
      2
       
    </div>
  </span>
  <span
    className="ReflectionListItem_physicalcontent"
  >
    <div>
       physical content:  
      stuff
        
    </div>
  </span>
  <button
    onClick={[Function]}
  >
    Delete
  </button>
  <button
    onClick={[Function]}
  >
    Edit
  </button>
</div>
`;
